simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp

```diff
@@ -326,6 +347,34 @@ void SimplePlanningSimulator::set_initial_state(
   is_initialized_ = true;
 }
 
+double SimplePlanningSimulator::get_z_pose_from_trajectory(const double x, const double y)
+{
+  // calculate closest point on trajectory
+  if (!current_trajectory_ptr_) {
+    return 0.0;
+  }
+
+  const double max_sqrt_dist = std::numeric_limits<double>::max();
+  double min_sqrt_dist = max_sqrt_dist;
+  size_t index;
+  bool found = false;
+  for (size_t i = 0; i < current_trajectory_ptr_->points.size(); ++i) {
+    const double dist_x = (current_trajectory_ptr_->points.at(i).pose.position.x - x);
+    const double dist_y = (current_trajectory_ptr_->points.at(i).pose.position.y - y);
+    double sqrt_dist = dist_x * dist_x + dist_y * dist_y;
+    if (sqrt_dist < min_sqrt_dist) {
+      min_sqrt_dist = sqrt_dist;
+      index = i;
+      found = true;
+    }
+  }
+    if (found) {
+      return current_trajectory_ptr_->points.at(index).pose.position.z;
+    }
```

Comment:
clang format is not applied?

Reply:
(We can automatically apply clang-format-12 by pre-commit)

Reply:
We do not apply format to the packages copied from .auto because it becomes difficult to sync.

Reply:
(The source codes in .auto are not originally formatted. If we apply format, a lot of code differences will occur. )

Reply:
@tkimura4 
Ahh that makes sense, I didn't know that.
So you mean when porting behavior_velocity_planner, I shouldn't use clang format but uncrustify?

Reply:
@TakaHoribe 
↑

Reply:
Got it!!